### PR TITLE
fix(dependencies): changed the default http client to be more robust

### PR DESCRIPTION
**Pull request by sauren-khosla**
### Done checklist
- [ ] docs/SPEC.md updated
- [ ] Test cases written

### CODE REVIEW

File: dependencies/dependencies.go [hunk 1]
```diff
@@ -50,12 +52,29 @@ func (d Dependencies) URLValidator() (url.Validator, error) {
...
+		DialContext: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+			// DualStack is deprecated
+		}).DialContext,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+	}
```
Comment (nathanielc) [inline dependencies/dependencies.go]: I think you lost the max connections per host option.

File: dependencies/dependencies.go [hunk 2]
```diff
@@ -50,12 +53,30 @@ func (d Dependencies) URLValidator() (url.Validator, error) {
...
+		DialContext: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+			// DualStack is deprecated
+		}).DialContext,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+		MaxIdleConnsPerHost:   runtime.GOMAXPROCS(0) + 1,
```
Comment (nathanielc) [inline dependencies/dependencies.go]: What is the reasoning for this default? You had it set to `100` earlier.

Comment (sauren-khosla) [reply dependencies/dependencies.go]: I pulled this from a commit on another branch, but I just checked the defaults in a doc elsewhere and they changed to 100. I'll update that now.

File: dependencies/dependencies.go [hunk 3]
```diff
@@ -50,12 +52,30 @@ func (d Dependencies) URLValidator() (url.Validator, error) {
...
+		DialContext: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+			// DualStack is deprecated
+		}).DialContext,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+		MaxIdleConnsPerHost:   100,
```
Comment (nathanielc) [inline dependencies/dependencies.go]: Looks good, but can you add a comment to this saying is not part of the defaults so we know which ones we have changed and which are from Go's defaults?

**Pull request merged**